### PR TITLE
Insert processed tasks

### PR DESCRIPTION
**Pull request by velmer**
**Purpose of this PR**:
Insert tasks retrieved from another SAPS instance with proper ImageTaskState.

**What problem will be solved:**
Retrieved tasks from another SAPS instance can now be inserted on local ServiceCatalog.

**Depends on PR(s)**: #2

### CODE REVIEW

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcherImpl.java [hunk 1]
```diff
@@ -164,8 +166,10 @@ public void listTasksInDB() throws SQLException, ParseException {
                 	LOGGER.debug("----------------------------------------> " +  dataset);
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionDispatcherImpl.java]: as commented in the first PR, improve this log line.

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionManagerImpl.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return tasks;
+    }
+
+    /**
+     * Returns the URL of a SAPS remote instance.
+     *
+     * @return URL of a SAPS remote instance.
+     */
+    private String getRemoteInstanceUrl() {
+        return REMOTE_INSTANCE_URL;
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionManagerImpl.java]: it seems this should evolve in the future to a collection of URLs, right? (no need to change it now)

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionParameters.java [hunk 3]
```diff
@@ -0,0 +1,250 @@
+package org.fogbowcloud.saps.engine.core.dispatcher;
+
+import java.util.Date;
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionParameters.java]: the same comment as before, why do we need these methods?

File: src/main/java/org/fogbowcloud/saps/engine/core/model/ImageTask.java [hunk 4]
```diff
@@ -66,6 +66,27 @@ public ImageTask(String taskId, String dataset, String region, Date imageDate,
 		this.error = error;
 	}
 
+	public ImageTask(JSONObject imageTaskJsonObject) throws JSONException {
```
Comment (thiagomanel) [inline src/main/java/org/fogbowcloud/saps/engine/core/model/ImageTask.java]: or, forward the call to this(imageTaskJsonObject.getString("taskId"), imageTaskJsonObject.getString("dataset") ..)

**Review by thiagomanel — CHANGES_REQUESTED** [review]
thanks, @velmer I believe we can evolve fast in this PR. please, address the questions

File: src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionManagerImpl.java (hunk 2 above)
Comment (velmer) [reply src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/SubmissionManagerImpl.java]: Exactly. To proper test the feature in the future I'll create another PR containing this change.

File: src/main/java/org/fogbowcloud/saps/engine/core/model/ImageTask.java (hunk 4 above)
Comment (velmer) [reply src/main/java/org/fogbowcloud/saps/engine/core/model/ImageTask.java]: Agree

**Review by thiagomanel — APPROVED** [review]
thanks @velmer

**Pull request merged**